services/web/server/src/simcore_service_webserver/projects/_wallets_handlers.py

```diff
@@ -0,0 +1,107 @@
+""" Handlers for CRUD operations on /projects/{*}/wallet
+
+"""
+
+import functools
+import logging
+
+from aiohttp import web
+from models_library.api_schemas_webserver.wallets import WalletGet
+from models_library.projects import ProjectID
+from models_library.wallets import WalletGetDB, WalletID
+from pydantic import BaseModel, Extra
+from servicelib.aiohttp.requests_validation import parse_request_path_parameters_as
+from servicelib.aiohttp.typing_extension import Handler
+
+from .._meta import API_VTAG
+from ..login.decorators import login_required
+from ..security.decorators import permission_required
+from ..wallets import _api as wallet_api
+from ..wallets.exceptions import WalletAccessForbiddenError
+from . import projects_api
+from ._common_models import ProjectPathParams, RequestContext
+from .db import ProjectDBAPI
+from .exceptions import ProjectNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+
+def _handle_project_wallet_exceptions(handler: Handler):
+    @functools.wraps(handler)
+    async def wrapper(request: web.Request) -> web.StreamResponse:
+        try:
+            return await handler(request)
+
+        except ProjectNotFoundError as exc:
+            raise web.HTTPNotFound(reason=f"{exc}") from exc
+
+        except WalletAccessForbiddenError as exc:
+            raise web.HTTPForbidden(reason=f"{exc}") from exc
+
+    return wrapper
+
+
+routes = web.RouteTableDef()
+
+
+@routes.get(f"/{API_VTAG}/projects/{{project_id}}/wallet", name="get_project_wallet")
+@login_required
+@permission_required("project.wallet.*")
+@_handle_project_wallet_exceptions
+async def get_project_wallet(request: web.Request):
+    db: ProjectDBAPI = ProjectDBAPI.get_from_app_context(request.app)
+    req_ctx = RequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(ProjectPathParams, request)
+
+    # ensure the project exists
+    await projects_api.get_project_for_user(
+        request.app,
+        project_uuid=f"{path_params.project_id}",
+        user_id=req_ctx.user_id,
+        include_state=False,
+    )
+
+    wallet_db: WalletGetDB | None = await db.get_project_wallet(
+        project_uuid=path_params.project_id
+    )
+    output: WalletGet | None = WalletGet(**wallet_db.dict()) if wallet_db else None
```

Comment:
Also I am wondering a bit on the fact that get_project_wallet returns None. are we not usually raising an exception in that case? I think this would simplify a bit your code.

Reply:
to be honest I don't know what is better here. If the project does not have an assigned wallet is a valid use case so in my mind it does not need to be an exception. Do you think it should be an exception even though it is a valid case that there is no wallet?